docs/next/reference/buffer_format.rst

```diff
@@ -0,0 +1,283 @@
+.. _buffer-format-label:
+
+Buffer Format
+=============
+
+.. py:module:: moderngl
+.. py:currentmodule:: moderngl
+
+Description
+-----------
+
+A buffer format is a short string describing the layout of data in a vertex
+buffer array (VBO).
+
+A VBO often contains a homogeneous array of C-like structures. The buffer
+format describes what each element of the array looks like. For example,
+a buffer containing an array of high-precision 2D vertex positions might have
+the format ``"2f8"`` - each element of the array consists of two floats, each
+float being 8 bytes wide, ie. a double.
+
+Buffer formats are used in the :py:meth:`Context.vertex_array()` constructor,
+as the 2nd component of the `content` arg. See the :ref:`example-of-simple-usage-label` below.
+
+Syntax
+------
+
+A buffer format looks like:
+
+    ``[count]type[size] [[count]type[size]...] [/usage]``
+
+Where:
+
+- ``count`` is an optional integer. If omitted, it defaults to ``1``.
+- ``type`` is a single character indicating the data type:
+
+   - ``f`` float
+   - ``i`` int
+   - ``u`` unsigned int
+   - ``x`` padding
+- ``size`` is an optional number of bytes used to store the type.
+  If omitted, it defaults to 4 for numeric types, or to 1 for padding bytes.
+
+  A format may contain multiple, space-separated ``[count]type[size]`` triples
+  (See the :ref:`example-of-single-interleaved-array-label`), followed by:
+
+
+- ``/usage`` is optional. It may be preceded by a space, and then consists of
```

Comment:
may -> should

In moderngl 6 it will be a must. It caused a confusion because on the first sight one may think it belongs only to the last element.

Reply:
fixed